### PR TITLE
Improve performance of local builds

### LINKED ISSUE #2568: Improve performance of local builds
### Description

Local builds seem to involve a lot of rebuilding each time.

TODO:
- [ ] Check if the MSBuild configuration is correct for EmbeddedVsix / Integration.Vsix and if we're not doing unnecessary rebuilds.

### PR DESCRIPTION
**Pull request by rita-sonarsource**
Fixes #2568

### CODE REVIEW

**Review by duncanp-sonar — APPROVED** [review]

File: src/EmbeddedVsix/EmbeddedVsix.csproj [hunk 1]
```diff
@@ -102,6 +103,9 @@
     <!-- Specifies the target inputs and outputs so MSBuild can skip fetching the files if the analyzer version has not changed. -->
     <AdditionalFilesInput Include="..\EmbeddedSonarAnalyzer.props" />
     <AdditionalFilesOutput Include="$(OutputPath)$(AssemblyName).vsix" />
+    
+    <!-- Prevent the vsix from being cleaned during incremental builds -->
+    <FileWritesShareable Include="$(OutputPath)$(AssemblyName).vsix"/>
```
[the remark this answers is not in the document]
Comment (duncanp-sonar) [reply src/EmbeddedVsix/EmbeddedVsix.csproj:108]: We're trying to work round a bug in the SDK targets, so whatever we do will be a bit hacky. The best we can do is make it obvious what the bug is and what we have in our target files to work round it (FYI I checked and the same problem exists in a "clean" new VSIX that doesn't contain all of our extra targets and tweaks - one of the files is always regenerated, which in turn means the VSIX is always re-created).

We could make our hacks more obvious by co-locating them in the file as much as possible e.g.

```
<!-- *********************************************************-->
<!-- Hack to work round the VSIX being created every time -->
<!-- *********************************************************-->
<PropertyGroup>
  <CreateVsixContainer>false</CreateVsixContainer>
</PropertyGroup>

<ItemGroup>
    <!-- Prevent the vsix from being cleaned during incremental builds -->
    <FileWritesShareable Include="$(OutputPath)$(AssemblyName).vsix"/>
</ItempGroup>

<Target Name="SonarLint_ForceCreateVsix" ... />
<!-- *********************************************************-->

```

**Review by duncanp-sonar — APPROVED** [review]
I think we should try to locate the changes together as much as possible. Other than that, looks good.

File: src/EmbeddedVsix/EmbeddedVsix.csproj [hunk 2]
```diff
@@ -102,6 +129,9 @@
     <!-- Specifies the target inputs and outputs so MSBuild can skip fetching the files if the analyzer version has not changed. -->
     <AdditionalFilesInput Include="..\EmbeddedSonarAnalyzer.props" />
     <AdditionalFilesOutput Include="$(OutputPath)$(AssemblyName).vsix" />
+    
+    <!-- Prevent the vsix from being cleaned during incremental builds -->
```
Comment (duncanp-sonar) [inline src/EmbeddedVsix/EmbeddedVsix.csproj:133]: Shouldn't this property-set have been deleted? (now done in line 17)

Comment (rita-sonarsource) [reply src/EmbeddedVsix/EmbeddedVsix.csproj:133]: You're right, it's a conflict, I'll fix and force push

**Pull request merged**